src/client/libds3/ds3_internal.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * (C) Copyright 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#define D_LOGFAC	DD_FAC(client)
+
+#ifndef __DAOS_S3_INTERNAL_H__
+#define __DAOS_S3_INTERNAL_H__
+
+#include "daos.h"
+#include "daos_s3.h"
+#include "daos_fs.h"
+
+#define METADATA_BUCKET		"_METADATA"
+#define MULTIPART_MAX_PARTS	10000
+#define LATEST_INSTANCE		"latest"
+#define LATEST_INSTANCE_SUFFIX	"[latest]"
+
+#define METADATA_DIR_LIST			\
+	X(USERS_DIR, "users")			\
+	X(EMAILS_DIR, "emails")			\
+	X(ACCESS_KEYS_DIR, "access_keys")	\
+	X(MULTIPART_DIR, "multipart")
+
+/* Define for RPC enum population below */
+#define X(a, b) a,
+enum metadata_dirs {
+	METADATA_DIR_LIST
+	METADATA_DIR_LAST
```

Comment:
What does this do?

Reply:
It defines an enum with USERS_DIR, EMAILS_DIR, ...
That's a C trick we use in various DAOS places to avoid writing a lot of duplicated code.